### PR TITLE
fix: Send email when change in session state

### LINKED ISSUE #7155: Session accept reject email sent without state changes
In session after_update_object, session accept reject email is sent if the final state of session is 'accepted' or 'rejected'. This is not desirable as the email will be sent even when there is any update about the session and not necessarily change in state. Thus, email should only be sent when there a change in 'state' of session, and not any change in session data itself

### PR DESCRIPTION
**Pull request by Haider8**
Fixes #7155 

#### Short description of what this resolves:
Email was being sent only when the final session state is accepted or rejected.

#### Changes proposed in this pull request:
Send email when there's a change in session state.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ app/api/sessions.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5899038&bid=19316143)

**Comment by iamareebjamal** [comment]
That's even worse. Read the issue again

**Comment by Haider8** [comment]
We want to send email if there's any change in session details. Then we won't be checking if there's `'state' in data`?

**Comment by iamareebjamal** [comment]
That's opposite of what we want. Why will we want that? Read the issue again carefully

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/api/sessions.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5899038&bid=19316143)

File: app/api/sessions.py [hunk 1]
```diff
@@ -256,6 +256,13 @@ def before_update_object(self, session, data, view_kwargs):
             )
 
         new_state = data.get('state')
+
+        g.send_email = (
+            new_state is not None
+            and new_state != session.state
+            and (new_state == 'accepted' or new_state == 'rejected')
+        )
```
Comment (iamareebjamal) [inline app/api/sessions.py:264]: This should be written inside the if condition below

Comment (iamareebjamal) [reply app/api/sessions.py:264]: This also does not contain the data.get('send_email') logic

Comment (Haider8) [reply app/api/sessions.py:264]: So below I should write `(new_state == 'accepted' or new_state == 'rejected')` and `data.get('send_email', None)` apart from `g.send_email`?

Comment (iamareebjamal) [reply app/api/sessions.py:264]: Yes

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ app/api/sessions.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5899038&bid=19316143)

**Pull request merged**